tests/unittests/tests-hashes/tests-hashes-sha512.c

```diff
@@ -0,0 +1,417 @@
+    TEST_ASSERT(calc_and_compare_hash_wrapper(sdigits_letters, hdigits_letters));
+}
+
+static void test_hashes_sha512_hash_sequence_pangramm(void)
+{
+    TEST_ASSERT(calc_and_compare_hash(spangramm, hpangramm));
+    TEST_ASSERT(calc_and_compare_hash_wrapper(spangramm, hpangramm));
+}
+
+static void test_hashes_sha512_hash_sequence_pangramm_no_more(void)
+{
+    TEST_ASSERT(calc_and_compare_hash(spangramm_no_more, hpangramm_no_more));
+    TEST_ASSERT(calc_and_compare_hash_wrapper(spangramm_no_more, hpangramm_no_more));
+}
+
+static void test_hashes_sha512_hash_sequence_empty(void)
+{
+    TEST_ASSERT(calc_and_compare_hash(sempty, hempty));
+    TEST_ASSERT(calc_and_compare_hash_wrapper(sempty, hempty));
+}
+
+static void test_hashes_sha512_hash_sequence_failing_compare(void)
+{
+    /* failing compare (switched last byte of expected hash from `a4` to `a5`) */
+    TEST_ASSERT(!calc_and_compare_hash(sfailing_compare, hfailing_compare));
+    TEST_ASSERT(!calc_and_compare_hash_wrapper(sfailing_compare, hfailing_compare));
+}
+
+static void test_hashes_sha512_hash_long_sequence(void)
+{
+    TEST_ASSERT(calc_and_compare_hash(slong_sequence, hlong_sequence));
+    TEST_ASSERT(calc_and_compare_hash_wrapper(slong_sequence, hlong_sequence));
+}
+
+static void test_hashes_sha512_hash_sequence_abc(void)
+{
+    TEST_ASSERT(calc_and_compare_hash(s_fips_oneblock, h_fips_oneblock));
+    TEST_ASSERT(calc_and_compare_hash_wrapper(s_fips_oneblock, h_fips_oneblock));
+}
+
+static void test_hashes_sha512_hash_sequence_abc_long(void)
+{
+    TEST_ASSERT(calc_and_compare_hash(s_fips_multiblock, h_fips_multiblock));
+    TEST_ASSERT(calc_and_compare_hash_wrapper(s_fips_multiblock, h_fips_multiblock));
+}
+
+static void test_hashes_sha512_hash_sequence_binary(void)
+{
+    static uint8_t hash[SHA512_DIGEST_LENGTH];
+    sha512(s_coffe, sizeof(s_coffe), hash);
+    TEST_ASSERT(memcmp(h_coffee, hash, SHA512_DIGEST_LENGTH) == 0);
+}
+
+static void test_hashes_sha512_hash_update_twice(void)
+{
+    static const char *teststring = "abcdef";
+
+    static uint8_t hash_update_once[SHA512_DIGEST_LENGTH];
+    static uint8_t hash_update_twice[SHA512_DIGEST_LENGTH];
+    sha512_context_t sha512;
+
+    sha512_init(&sha512);
+    sha512_update(&sha512, (uint8_t*)teststring, sizeof(teststring));
+    sha512_final(&sha512, hash_update_once);
+
+    sha512_init(&sha512);
+    sha512_update(&sha512, (uint8_t*)teststring, 3);
+    sha512_update(&sha512, (uint8_t*)&teststring[3], sizeof(teststring)-3);
+    sha512_final(&sha512, hash_update_twice);
+
+    TEST_ASSERT(memcmp(hash_update_once, hash_update_twice, SHA512_DIGEST_LENGTH) == 0);
+}
+
+static void test_hashes_sha512_hash_static_buffer(void)
+{
+    void *digest = NULL;
+    digest = sha512(s_fips_oneblock, strlen(s_fips_oneblock), NULL);
+    TEST_ASSERT(digest != NULL && memcmp(h_fips_oneblock, digest, SHA512_DIGEST_LENGTH) == 0);
+    digest = NULL;
```

Comment:
Why? Shouldn't the address stay the same since it is static?

Reply:
Removed it now since it is written anyway in the next line.